packages/fx-core/src/core/environment.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  ConfigFolderName,
+  CryptoProvider,
+  err,
+  FxError,
+  Json,
+  ok,
+  Result,
+} from "@microsoft/teamsfx-api";
+import path from "path";
+import fs from "fs-extra";
+import { deserializeDict, dataNeedEncryption, mergeSerectData, PathNotExistError } from "..";
+
+export interface EnvInfo {
+  envName: string;
+  data: Json;
+}
+
+export interface EnvFiles {
+  envProfile: string;
+  userDataFile: string;
+}
+
+class EnvironmentManager {
+  private readonly defaultEnvName = "default";
+
+  public async loadEnvProfile(
+    projectPath: string,
+    envName?: string,
+    cryptoProvider?: CryptoProvider
+  ): Promise<Result<EnvInfo, FxError>> {
+    if (!(await fs.pathExists(projectPath))) {
+      return err(PathNotExistError(projectPath));
+    }
+
+    envName = envName ?? this.defaultEnvName;
+    const envFiles = this.getEnvFilesPath(envName, projectPath);
+    const userDataResult = await this.loadUserData(envFiles.userDataFile, cryptoProvider);
+    if (userDataResult.isErr()) {
+      return err(userDataResult.error);
+    }
+
+    const userData = userDataResult.value;
+    const data = await fs.readJson(envFiles.envProfile);
```

Comment:
>fs.readJson(envFiles.envProfile); [](http://example.com/codeflow?start=23&length=33)

We'd better check the `env.${envName}.json` file before reading it. We can have a pre-validation to check the envirronment files, and:
1. Implicitly create the file(s) if it is missing (may be deleted by user by accident);
**OR**
2. Pop up error if env files not found.

Reply:
For this stage(M1), I would prefer to keep the user behavior unchanged, but it's valuable to handle it gracefully. Let's add it as TODO comment and take care of it later.
Agree that we do need to return accurate error if the `env.${envName}.json` file doesn't exist for further error handling.